packages/block-library/src/image/editor.scss

```diff
@@ -61,6 +61,12 @@ figure.wp-block-image:not(.wp-block) {
 		display: inline;
 	}
 
+	// Shown while image is being uploded but cannot be previewed.
+	.block-editor-image__skeleton {
+		aspect-ratio: 4 / 3;
+		background: $gray-100;
+	}
+
```

Comment:
Would it be worth reusing the placeholder state for when an image is deselected? Currently that placeholder state uses a transparent background, and a different size, so with the gray color and 4/3, I notice a flash of gray and a different height during the upload:

https://github.com/user-attachments/assets/fa769845-a91c-4564-a6dc-86861562d80e

It's very quick in the video, so might be easy to miss 😄

Reply:
Deselected!! I knew we already had a placeholder state, but couldn't find it 😛 Good suggestion I'll do that.

Reply:
I've switched to using `Placeholder`.

I'm not a big fan of how the spinner looks on top of the diagonal line, though. It makes the spinner hard to see.

<img width="662" alt="Screenshot 2024-07-18 at 14 47 31" src="https://github.com/user-attachments/assets/3fcbd5eb-6d79-4b6f-ac8b-07e5be2dc332">

Mind you, this problem also exists on `trunk`. When there is a preview it can be hard to see the spinner depending on the image being uploaded.

<img width="660" alt="Screenshot 2024-07-18 at 14 48 52" src="https://github.com/user-attachments/assets/8a2a0d91-8938-42b7-9ba2-962c599a8459">

Maybe we should put the spinner in at translucent box?

<img width="678" alt="Screenshot 2024-07-18 at 14 47 22" src="https://github.com/user-attachments/assets/acf98e19-b0f2-4131-967b-68f32d6cda0b">

<img width="641" alt="Screenshot 2024-07-18 at 14 51 43" src="https://github.com/user-attachments/assets/d4adde27-b24d-4ef8-b8d3-ce2bd113a217">

Or maybe we should denote that the image is uploading using a different technique e.g. pulsating the block's opacity?

Or maybe it's fine how it is?

What do you think @jasmussen?

Reply:
> What do you think @jasmussen?

I'm not sure if Joen is around at the moment, so I'll just give a soft ping to @WordPress/gutenberg-design for visibility 🙂

Reply:
Good plan. We can come back to this later, at any rate. HEIC is not very common and what's in this PR is better than what's in trunk.

Reply:
I'd lean towards tweaking the overlay color/opacity a bit, so that the spinner remains visible, rather than adding another smaller 'surface' behind the spinner.

Another option could be a different loading animation in place of the spinner. The overlay could pulse between different opacity levels. That might require more holistic exploration though, so identify other applications.

Reply:
Here's what increasing the opacity of the placeholder from 0.1 to 0.2 looks like against a few different backgrounds:

<img width="1132" alt="Screenshot 2024-07-19 at 15 09 46" src="https://github.com/user-attachments/assets/d812105e-c52a-43c4-a938-fcd00a03d8fe">
<img width="1149" alt="Screenshot 2024-07-19 at 15 09 55" src="https://github.com/user-attachments/assets/27785a3f-a84f-42c6-b187-0abbf30cc766">
<img width="1123" alt="Screenshot 2024-07-19 at 15 10 05" src="https://github.com/user-attachments/assets/262d61b4-4f74-4aff-b7b3-1bad995d6cfd">

That OK with you @jameskoster? If you want to play with it yourself an easy way to test is to make this change so that the placeholder and spinner will aways show for all image blocks:

```diff
diff --git a/packages/block-library/src/image/image.js b/packages/block-library/src/image/image.js
index 385a740e652..3417b1fc1a8 100644
--- a/packages/block-library/src/image/image.js
+++ b/packages/block-library/src/image/image.js
@@ -827,7 +827,7 @@ export default function Image( {
 	const isRounded = attributes.className?.includes( 'is-style-rounded' );
 
 	let img =
-		temporaryURL && hasImageErrored ? (
+		true || ( temporaryURL && hasImageErrored ) ? (
 			// Show a placeholder during upload when the blob URL can't be loaded. This can
 			// happen when the user uploads a HEIC image in a browser that doesn't support them.
 			<Placeholder
```

Also happy to punt this to later.

Reply:
Thanks for that tip :) 

Does the overlay background need to be `currentColor`? Since it's part of the UI I think it could use a UI color like `$gray-900`. Then with opacity something like .7 we get this:

<img width="985" alt="Screenshot 2024-07-19 at 10 18 27" src="https://github.com/user-attachments/assets/d73989af-1b7f-4281-b04c-f8f685b4ac0d">

This ensures the spinner is visible, and has similar visual weight to the dropzone treatment. 

What do you think?

Reply:
I like it but I'm scared that I'm hiding a UI change that needs some design attention in an otherwise innocuous PR 😅

I'll revert f5e8ac8, merge this, and then open a new PR with your suggestion.

Reply:
https://github.com/WordPress/gutenberg/pull/63885